modules/gke-hub/main.tf

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "google_gke_hub_membership" "membership" {
+  provider      = google-beta
+  for_each      = var.member_clusters
+  membership_id = each.key
+  project       = var.project_id
+  endpoint {
+    gke_cluster {
+      resource_link = each.value
+    }
+  }
+}
+
+resource "google_gke_hub_feature" "feature-configmanagement" {
+  provider = google-beta
+  for_each = var.features.configmanagement ? { 1 = 1 } : {}
+  project  = var.project_id
+  name     = "configmanagement"
+  location = "global"
+}
+
+resource "google_gke_hub_feature" "feature-mci" {
+  provider = google-beta
+  for_each = var.features.mc_ingress ? { 1 = 1 } : {}
+  project  = var.project_id
+  name     = "multiclusteringress"
+  location = "global"
+  spec {
+    multiclusteringress {
+      config_membership = google_gke_hub_membership.membership[each.key].id
+    }
+  }
+}
+
+resource "google_gke_hub_feature" "feature-mcs" {
+  provider = google-beta
+  for_each = var.features.mc_servicediscovery ? { 1 = 1 } : {}
+  project  = var.project_id
+  name     = "multiclusterservicediscovery"
+  location = "global"
+}
+
+resource "google_gke_hub_feature_membership" "feature_member" {
+  provider   = google-beta
+  for_each   = var.member_clusters
+  project    = var.project_id
+  location   = "global"
+  feature    = google_gke_hub_feature.feature-configmanagement["1"].name
+  membership = google_gke_hub_membership.membership[each.key].membership_id
+
+  dynamic "configmanagement" {
+    for_each = (
+      try(var.member_features.configmanagement, null) != null
```

Comment:
I the var becomes non nullable, this can be a simple coalesce.

Reply:
done

Reply:
for some reason coalesce doesn't work here
```
│ Error: Error in function call
│
│   on ../../cloud-foundation-fabric/modules/gke-hub/main.tf line 67, in resource "google_gke_hub_feature_membership" "feature_member":
│   67:     for_each = coalesce(var.member_features.configmanagement, {})
│     ├────────────────
│     │ var.member_features.configmanagement is a object, known only after apply
```